oshmem/mca/sshmem/ucx/sshmem_ucx_module.c

```diff
@@ -180,18 +180,27 @@ segment_create(map_segment_t *ds_buf,
     unsigned flags       = UCP_MEM_MAP_ALLOCATE;
     int status           = OSHMEM_SUCCESS;
 
-#if HAVE_DECL_UCS_MEMORY_TYPE_RDMA
     if (hint & SHMEM_HINT_DEVICE_NIC_MEM) {
+#if HAVE_DECL_UCS_MEMORY_TYPE_RDMA
         status = segment_create_internal(ds_buf, NULL, size, flags,
                                          UCS_MEMORY_TYPE_RDMA);
         if (status == OSHMEM_SUCCESS) {
             ds_buf->alloc_hints = hint;
             ds_buf->allocator   = &sshmem_ucx_allocator;
             return OSHMEM_SUCCESS;
+        } else {
+            SSHMEM_WARN("Failed to allocate DEVICE_NIC_MEM, "
```

Comment:
1. Move this error message into segment_create_internal, right after the call to ucp_mem_map.
2. Add ucs_status_string(status) to the error message

Reply:
There is already a SSHMEM_ERROR there can I just merge them into a single message?

Reply:
> There is already a SSHMEM_ERROR there can I just merge them into a single message?

Yes

Reply:
@yosefe after the err msg was moved into `segment_create_internal`, we now have a msg with "DEVICE_NIC_MEM" text, in a generic flow, not specific to RDMA memtype, which is bad IMO.
I suggest we return the err msg back here, or at least need to remove the mentioning of "DEVICE_NIC_MEM" there. WDYT?